lib/widgets/components/dropdown.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/base/style/text_styles.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class Dropdown<T extends StateStreamable<K>, K, C> extends StatelessWidget {
+  final void Function(C?) onChanged;
+  final C value;
+  final List<DropdownMenuItem<C>> items;
+
+  Dropdown({required this.onChanged, required this.items, required this.value});
+
+  final underline = Container(
+    height: 1.0,
+    decoration: const BoxDecoration(
+      border: Border(
+        bottom: BorderSide(
+          color: AppColor.slightlyHighlighted,
+          width: 0.0,
+        ),
+      ),
+    ),
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<T, K>(
+      builder: (context, state) {
+        return DropdownButton<C>(
+          dropdownColor: AppColor.secondary,
+          underline: underline,
+          icon: const Icon(Icons.arrow_drop_down, color: AppColor.white),
+          style: AppTextStyle.buttonText,
+          value: value,
+          items: items,
```

Comment:
Nice work on this file.
Could `items` here also contain the heavy lifting done in `StatisticsDropdown`/`ReceiptsDropdown`? Seems like there's a bit of copy-paste work between the two.

Reply:
this one is very hard to solve because both StatisticsFilterCategory and ReceiptFilterCategoryboth implement the 'name' as an extension and is therefore unavailable in 'dropdown.dart'. As far as i can tell there is no way to implement an interface or a superclass for enums so the only solution as far as i know is to use dynamic types which may be a bit ugly. Can you have a look?

Reply:
I'll take a look at it

Reply:
Yes that is not good. I changed it back so generating the items is again the responsibility of each dropdown implementation